+hw/DaqController.m

```diff
@@ -167,18 +195,43 @@ function command(obj, varargin)
           end
           readyWait(obj);
           obj.DaqSession.release;
-        elseif any(~analogueChannelsIdx)
-            waveforms = waveforms(~analogueChannelsIdx);
-            for n = 1:length(waveforms)
-              digitalValues = waveforms{n};
-              for m = 1:length(digitalValues)
-                obj.DigitalDaqSession.outputSingleScan(digitalValues(m));
+        end
+        % for all digital or clock outputs (why does this have to be an else?)
+        if any(~analogueChannelsIdx)&&any(values(:,~analogueChannelsIdx)~=0)
+          maxLnWaveform = max(cellfun(@length, waveforms));
+          % pad shorter waveforms
+          for i = 1:length(waveforms)
+            waveforms{i}(end:maxLnWaveform) = waveforms{i}(end);
+          end
+          waveformsMtx = cell2mat(waveforms);
+          %if iscolumn(waveformsMtx), waveformsMtx = waveformsMtx'; end
+          % output columns of waveformsMtx (values for each channel)
+          for n = 1:size(waveformsMtx,2)
+            %if we have some value to output
+            if any(waveformsMtx(:,n))
+              % for clock output channels with a valid value to output
+              if strncmp('ctr',(obj.DaqChannelIds{n}),3)
+                obj.ClockDaqSession.DurationInSeconds = length(waveformsMtx) / obj.SampleRate;
+                %Duty Cycle must be b/w 0-1, so set to 'n' and scale frequency by 1/n
+                obj.ClockDaqSession.Channels.DutyCycle = 0.99;
+                obj.ClockDaqSession.Channels.Frequency = 1/obj.ClockDaqSession.DurationInSeconds/0.99;
+                startBackground(obj.ClockDaqSession);
+                %for digital output channels
+              elseif strncmp('port',(obj.DaqChannelIds{n}),4) 
+                obj.DigitalDaqSession.outputSingleScan(waveformsMtx(:,n));
               end
             end
+          end
         end
       end
     end
     
+    function clearSessions(obj)
```

Comment:
Although this is fine for debugging, it's probably not worth keeping in the code.  Instead, you could have the createDaqChannels method check whether a channel of that name as already been added before attempting to add one.